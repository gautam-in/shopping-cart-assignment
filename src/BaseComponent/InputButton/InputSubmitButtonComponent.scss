/******************* InputSubmitButtonComponent *****************************/
@import '../../Styles/variable.scss';
@import '../../Styles/color-variable.scss';

$BUTTON_PADDING: 15px;

.input-button {
    border: none;
    background-color: $CRIMSON_RED;
    width: 100%;
    min-height: $MEDIUM_HEIGHT;
    padding: $BUTTON_PADDING;
    font-size: $MEDIUM_FONT_SIZE;
    color: $WHITE;
}
