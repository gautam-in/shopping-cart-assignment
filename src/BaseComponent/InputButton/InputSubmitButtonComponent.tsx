import "./InputSubmitButtonComponent.scss";

interface IInputSubmitButtonComponentProps {
  id: string;
  buttonName: string;
  externalClassName?: string;
}

export const InputSubmitButtonComponent: React.FC<IInputSubmitButtonComponentProps> = (
  props
) => {
  const { id, buttonName, externalClassName } = props;
  const BASE_CLASSNAME: string = "input-button";
  return (
    <div className={externalClassName}>
      <input 
        id={id} 
        className={BASE_CLASSNAME}
        type="submit"
        value={buttonName}
        />
    </div>
  );
};
