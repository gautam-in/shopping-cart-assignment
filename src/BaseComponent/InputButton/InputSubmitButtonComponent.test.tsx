import { ShallowWrapper, shallow } from "enzyme";
import { InputSubmitButtonComponent } from "./InputSubmitButtonComponent";

describe("InputSubmitButtonComponent", () => {
    const externalClassName: string = "input-button"
    const tree: ShallowWrapper = shallow(
        <InputSubmitButtonComponent id="button" buttonName="Submit" externalClassName={externalClassName}/>
    );

    test("InputSubmitButtonComponent Snapshot render", () => {
        expect(tree).toMatchSnapshot();
    });

    test("Check InputSubmitButtonComponent has classname", () => {
        expect(tree.hasClass(externalClassName)).toEqual(true)
    });
});
