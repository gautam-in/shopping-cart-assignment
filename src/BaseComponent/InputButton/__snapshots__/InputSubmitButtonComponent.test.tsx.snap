// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSubmitButtonComponent InputSubmitButtonComponent Snapshot render 1`] = `ShallowWrapper {}`;
